misc/tutorial/218_columns_filters.ngdoc

```diff
@@ -0,0 +1,236 @@
+            filter: {
+              condition: uiGridConstants.filter.ENDS_WITH,
+              placeholder: 'ends with'
+            }, headerCellClass: $scope.highlightFilteredHeader
+          },
+          // custom condition function
+          {
+            field: 'phone',
+            filter: {
+              condition: function(searchTerm, cellValue) {
+                var strippedValue = (cellValue + '').replace(/[^\d]/g, '');
+                return strippedValue.indexOf(searchTerm) >= 0;
+              }
+            }, headerCellClass: $scope.highlightFilteredHeader
+          },
+          // multiple filters
+          { field: 'age',
+            columnFilter: {
+              type: 'number'
+            }
+
+          , headerCellClass: $scope.highlightFilteredHeader},
+          // date filter
+          {
+          field: 'mixedDate',
+          cellFilter: 'date',
+          width: '15%', filter: {
+              condition: uiGridConstants.filter.LESS_THAN,
+              placeholder: 'less than',
+              term: nextWeek,
+              type: 'date'
+            }, columnFilter: {dateType: 'datetime-local'},
+            headerCellClass: $scope.highlightFilteredHeader
+          },
+          { field: 'mixedDate', displayName: "Long Date", cellFilter: 'date:"longDate"', filterCellFiltered:true, width: '15%',
+          }
+        ]
+      };
+
+      $http.get('/data/500_complex.json')
+        .success(function(data) {
+          $scope.gridOptions.data = data;
+          $scope.gridOptions.data[0].age = -5;
+
+          data.forEach( function addDates( row, index ){
+            row.mixedDate = new Date();
+            row.mixedDate.setDate(today.getDate() + ( index % 14 ) );
+            row.gender = row.gender==='male' ? '1' : '2';
+          });
+        });
+
+      $scope.toggleFiltering = function(){
+        $scope.gridOptions.enableFiltering = !$scope.gridOptions.enableFiltering;
+        $scope.gridApi.core.notifyDataChange( uiGridConstants.dataChange.COLUMN );
+      };
+    }])
+    .filter('mapGender', function() {
+      var genderHash = {
+        1: 'male',
+        2: 'female',
+        3: 'Unknown'
+      };
+
+      return function(input) {
+        if (!input){
+          return '';
+        } else {
+          return genderHash[input];
+        }
+      };
+    });
+  </file>
+  <file name="index.html">
+    <div ng-controller="MainCtrl">
+      You can use asterisks to fuzz-match, i.e. use "*z*" as your filter to show any row where that column contains a "z".
+      <br>
+      <br>
+      <strong>Note:</strong> The third column has the filter input disabled, but actually has a filter set in code that requires every company to have an 'a' in their name.
+      <br>
+      <br>
+      <button id='toggleFiltering' ng-click="toggleFiltering()" class="btn btn-success">Toggle Filtering</button>
+      <div id="grid1" ui-grid-columns-filters-directive ui-grid="gridOptions" class="grid"></div>
```

Comment:
`ui-grid-columns-filters-directive` This isn't the naming convention we follow for directive names on our features.